devbox/provision/provision.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+readonly GRAPL_ROOT="${PWD}"
+readonly GRAPL_DEVBOX_DIR="${HOME}/.grapl_devbox"
+mkdir -p "${GRAPL_DEVBOX_DIR}"
+
+########################################
+# Helper functions
+########################################
+has_key() {
+    local -r input_json="${1}"
+    local -r key="${2}"
+    jq --exit-status ".[\"${key}\"]" <<< "${input_json}" > /dev/null
+}
+
+########################################
+# Ensure we're in the Pulumi-enabled venv
+########################################
+if [ ! -f build-support/venv/bin/activate ]; then
+    echo "Set up your virtualenv with 'build-support/manage_virtualenv.sh'"
+    exit 42
+fi
+# shellcheck disable=SC1091
+source build-support/venv/bin/activate
+
+########################################
+# Set up an SSH key specifically for the Devbox
+########################################
+
+readonly SSH_PRIVATE_KEY_FILE="${GRAPL_DEVBOX_DIR}/ssh_key"
+readonly SSH_PUBLIC_KEY_FILE="${SSH_PRIVATE_KEY_FILE}.pub"
+
+if [ ! -f "${SSH_PRIVATE_KEY_FILE}" ]; then
+    ssh-keygen -t ed25519 -a 100 -C "${USER}@graplsecurity.com" -f "${SSH_PRIVATE_KEY_FILE}"
```

Comment:
Just FYI, not everybody's local account name matches their email address (I'm `maier` on one, `chris` on the other, for example).

Reply:
Yikes. I'm not really sure how to account for these variations?